src/couch/src/couch_server.erl

```diff
@@ -381,7 +369,7 @@ handle_config_change("couchdb_engines", _, _, _, N) ->
     {ok, N};
 handle_config_change("admins", _, _, Persist, N) ->
     % spawn here so couch event manager doesn't deadlock
-    spawn(fun() -> hash_admin_passwords(Persist) end),
+    couch_password_server:hash(true, Persist),
```

Comment:
I wonder if another problem here is that we'll be rehashing the password N times, once for each couch_server? For instance on a 96 cpu server, we'll rehash the password 96 times in parallel, it seems.

An ideal way to handle it might be to just remove the password rehashing from here to the new password hasher gen_server, which we have one of. So it would have the `handle_config_change` callback and so on. A quicker hack for now could be to only handle admin rehashing for `N=1`. Also do the same for httpd changes, we don't need to stop/restart the httpd backend N times.

Reply:
Is N the number of Nodes in the cluster or the Number of processors?

Reply:
Or can we use the `State` record to call to number of unhashed passwords and only start the hashing if it's greater than zero?

Reply:
`N` is the couch_server process number `1 ... $(number_of_schedulers)`. So in handle_config_change ... we'd replace `N` with `1`. So the first couch_server would do the hashing or stopping of httpd and the rest would ignore it.

Reply:
To debug this try to log N and the rest of the state and then in remsh do something like `config:set("admins", "adm5", "pass5").`

Reply:
So, I pattern-matched the call for the server with Number 1, do we need to handle the "admin" for other servers too?

Reply:
For others it can just be a fall-through and nothing will happen:

This case basically:

```erlang
handle_config_change(_, _, _, _, N) ->
    {ok, N}.
```

Reply:
Ok.